cterasdk/core/plans.py

```diff
@@ -143,3 +182,51 @@ def delete(self, name):
         except CTERAException as error:
             logging.getLogger().error("Plan deletion failed.")
             raise CTERAException('Plan deletion failed', error)
+
+
+class PlanAutoAssignPolicy(BaseCommand):
+
+    def get_policy(self):
+        """
+        Get plans auto assignment policy
+        """
+        return self._portal.execute('', 'getPlanAutoAssignmentRules')
+
+    def set_policy(self, rules, apply_default=None, default=None, apply_changes=True):
+        """
+        Set plans auto assignment policy
+
+        :param list[cterasdk.common.types.PolicyRule] rules: List of policy rules
+        :param bool,optional apply_default: If no match found, apply default plan. If not passed, the current config will be kept
+        :param str,optional default: Name of a plan to assign if no match found. Ignored unless the ``apply_default`` is set to ``True``
+        """
+        plans = [rule.value for rule in rules]
```

Comment:
I think you should also validate the "default" plan. Otherwise, you might crash later when trying to get it its baseObjectRef. In addition, I think you should do list(set(plans)) to remove duplicates

Reply:
Fixed.
`plans = {rule.value for rule in rules}`
`if default:`
`    plans.add(default)`
`plans = list(plans)`
`portal_plans = {plan.name: plan for plan in self._portal.plans.by_name(plans, ['baseObjectRef'])}`

Then, when obtaining the `baseObjectRef`, we retrieve it from the dictionary.
`policy.defaultPlan = portal_plans.get(default).baseObjectRef`

Reply:
👍